internal/cmd/log-file-metric-exporter/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"flag"
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/cluster-logging-operator/internal/pkg/symnotify"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net/http"
+	"os"
+	"regexp"
+)
+
+var (
+	verbosity int = 0
+	//Reference regexp https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/blob/master/lib/fluent/plugin/filter_kubernetes_metadata.rb#L56, https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/kubelet/dockertools/docker.go
+	//compile k8 logfilepathname pattern
+	kubernetesregexpCompiled = regexp.MustCompile(`.var.log.containers.([a-z0-9][-a-z0-9]*[a-z0-9])_([^_]+)_(.+)-([a-z0-9]{64})\.log$`)
+)
+
+const (
+	PodNameIndex = iota + 1
+	NamespaceIndex
+	ContainerNameIndex
+	DockerIndex
+	MatchLen
+)
+
+type FileWatcher struct {
```

Comment:
The filename is `main.go` , may be we can move this struct in a different file/package? with a package level method `NewFileWatcher()` instead of creating it in main?